GeneratorInterface/GenFilters/src/PythiaFilter.cc

```diff
@@ -57,20 +58,20 @@ bool PythiaFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSetup)
      
      for ( HepMC::GenEvent::particle_const_iterator p = myGenEvent->particles_begin();
 	   p != myGenEvent->particles_end(); ++p ) {
-       
-       rapidity = 0.5*log( ((*p)->momentum().e()+(*p)->momentum().pz()) / ((*p)->momentum().e()-(*p)->momentum().pz()) );
+       HepMC::FourVector mom = zboost((*p)->momentum());
```

Comment:
please move the requirements that don't depend on the z-boost so they are imposed before the boost calculation